src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -592,12 +592,17 @@ void MeshForwarder::HandleSentFrameToChild(const Mac::Frame &aFrame, otError aEr
         iter.Advance();
         mIndirectStartingChild = iter.GetChild();
 
-        if (aError == OT_ERROR_NONE)
+        switch (aError)
         {
+        case OT_ERROR_NONE:
             child->ResetIndirectTxAttempts();
-        }
-        else
-        {
+            break;
+
+        case OT_ERROR_CHANNEL_ACCESS_FAILURE:
+        case OT_ERROR_ABORT:
+            break;
```

Comment:
If we `break` here for the CCA and ABORT errors, the code below the `switch` will assume the frame was sent and moves to next fragment (if any) and we won't retx the failed frame.

How about this instead?  
```
switch (aError)
{
case OT_ERROR_NONE:
     child->ResetIndirectTxAttempts();
     break;

case OT_ERROR_NO_ACK:
	child->IncrementIndirectTxAttempts();

	// fall through

case OT_ERROR_CHANNEL_ACCESS_FAILURE:
case OT_ERROR_ABORT:

	 if (child->GetIndirectTxAttempts() < kMaxPollTriggeredTxAttempts) 
	 ...
```
For NO_ACK error we increase the attempt counter, for other errors we just try again. Possible danger with this is that if we keep getting ABORT or CCA errors, we can remain stuck on re-trying the same frame. 

Alternative is to keep the logic here as before (which would retry up to a max for any type of failure).

Reply:
Ah yes, silly mistake on my part.  I've updated the code as suggested.

Yes, this update will cause the message queue to get stuck if the radio continues to return CCA or ABORT errors.  At the same time, if the radio is truly returning these errors continuously, is there value in advancing the message queue?

Reply:
Sounds good. Thanks